spring-cloud-stream-tuple/src/main/java/org/springframework/cloud/stream/tuple/JsonStringToTupleConverter.java

```diff
@@ -62,7 +63,7 @@ else if (node.isNumber()) {
 					builder.addEntry(name, node.numberValue());
 				}
 				else {
-					builder.addEntry(name, node.asText());
```

Comment:
Can we add the same approach to `node.isObject()` as well? I think eventually the logic will get invoked as we call `convert(node.toString)` for `node.isObject` check.